packages/compiler/test/checker/model.ts

```diff
@@ -73,6 +78,27 @@ describe("compiler: models", () => {
     strictEqual(calls, 0);
   });
 
+  it("emit single error when there is an invalid ref in a templated type", async () => {
+    testHost.addCadlFile(
+      "main.cadl",
+      `
+        model A<T> {t: T, invalid: notValidType }
+
+        model Bar {
+          instance1: A<string>;
+          instance2: A<int32>;
+        }
+        `
+    );
+    const diagnostics = await testHost.diagnose("main.cadl");
+    expectDiagnostics(diagnostics, [
```

Comment:
does this test function limit the diagnostics to just this one, or could other diagnostics have been reported?

Reply:
this function does check that there is exactly those diagnostics. Will give a good error message if there is more or less with the list of all the diagnostics so you can resolve the other cases.